ProefProeveVeluwe/Assets/Scripts/Runtime/GPS/GPS.cs

```diff
@@ -1,98 +1,65 @@
 
-        /// <summary>
-        /// checks the device for location data
-        /// </summary>
-        private IEnumerator GPSLocation()
-        {
-            //check if location is enabled
-            if (!Input.location.isEnabledByUser) yield break;
-            Input.location.Start();
+    [SerializeField] private float _timer = 30f;
+    private float _currentTimer;
 
-            //wait until service initialize
-            while (Input.location.status == LocationServiceStatus.Initializing && _currentTimer > 0)
-            {
-                yield return new WaitForSeconds(1);
-                _currentTimer--;
-            }
+    public bool HasLocation = false;
 
-            // service didn't init in maxTimer sec
-            if (_currentTimer < 1)
-            {
-                GPSStatus.text = "Time Out";
-                yield break;
-            }
 
-            if (Input.location.status == LocationServiceStatus.Failed)
-            {
-                GPSStatus.text = "Unable to determine device location";
-                yield break;
-            }
+    public void Awake()
+    {
+        _currentTimer = _timer;
+        StartCoroutine(GPSLocation());
+    }
+    
+    private IEnumerator GPSLocation()
+    {
+        //check if location is enabled
+        if (!Input.location.isEnabledByUser) yield break;
+        Input.location.Start();
 
-            //Access granted
-            _updateValue++;
-            InvokeRepeating("UpdateGPSData", 0.5f, 1);
+        //wait until service initialize
+        while (Input.location.status == LocationServiceStatus.Initializing && _currentTimer > 0)
+        {
+            yield return new WaitForSeconds(1);
+            _currentTimer--;
         }
 
-        /// <summary>
-        /// Fills in text with the GPS Data 
-        /// </summary>
-        private void UpdateGPSData()
+        // service didn't init in maxTimer sec
+        if (_currentTimer < 1)
         {
-            if (Input.location.status == LocationServiceStatus.Running)
-            {
-                //acces granted to GPS values and it has been initialized
-                GPSStatus.text = "Running";
-                latitudeValue.text = Input.location.lastData.latitude.ToString();
-                longitudeValue.text = Input.location.lastData.longitude.ToString();
-                altitudeValue.text = Input.location.lastData.altitude.ToString();
-                horizontalAccuracyValue.text = Input.location.lastData.horizontalAccuracy.ToString();
-                timeStampValue.text = Input.location.lastData.timestamp.ToString();
-                updateRateValue.text = _updateValue.ToString();
-
-                return;
-            }
+            HasLocation = false;
+            _currentTimer = _timer;
+            yield break;
+        }
 
-            //service has been stopped
-            GPSStatus.text = "Stop";
+        if (Input.location.status == LocationServiceStatus.Failed)
```

Comment:
Will this ever run in the while loop we keep running while _currentTimer is bigger than zero after wards we will run the next if statement because _currentTimer is smaller than zero and we are checking if _currentTimer is smaller than 1